template/tree_v4.go

```diff
@@ -308,12 +316,19 @@ func (t *TreeV4) add(address patricia.IPv4Address, tag GeneratedType, matchFunc
 			}
 			parent.Right = newCommonParentNodeIndex
 		}
-		return countIncreased, t.nodes[newNodeIndex].TagCount, nil
+		return countIncreased, t.nodes[newNodeIndex].TagCount
 	}
 }
 
 // Delete a tag from the tree if it matches matchVal, as determined by matchFunc. Returns how many tags are removed
-func (t *TreeV4) Delete(address patricia.IPv4Address, matchFunc MatchesFunc, matchVal GeneratedType) (int, error) {
+// - use DeleteWithBuffer if you can reuse slices, to cut down on allocations
+func (t *TreeV4) Delete(address patricia.IPv4Address, matchFunc MatchesFunc, matchVal GeneratedType) int {
+	return t.DeleteWithBuffer(make([]GeneratedType, 0), address, matchFunc, matchVal)
```

Comment:
It really doesn't matter much, but I think it works to just pass `nil` in these cases instead of making a zero-length slice?

Reply:
ya good call - updated

Reply:
Could also be done in the other non-buffering methods, but I worry the caller will be expecting non-nil, so I'll keep it as is for now. If you're interested in reducing allocations, you can use the buffering versions